SocketRocket/SRWebSocket.m

```diff
@@ -534,6 +534,14 @@ - (void)_initializeStreams;
     
     CFStreamCreatePairWithSocketToHost(NULL, (__bridge CFStringRef)host, port, &readStream, &writeStream);
     
+    //If voip is set in UIBackgroundMode, set apropriate properties on read and write stream
+    NSArray* backgroundModes = [[[NSBundle mainBundle] infoDictionary] objectForKey:@"UIBackgroundModes"];
+    
+    if (backgroundModes && [backgroundModes containsObject:@"voip"]) {
+        CFReadStreamSetProperty(readStream, kCFStreamNetworkServiceType, kCFStreamNetworkServiceTypeVoIP);
```

Comment:
I agree that we shouldn't add new code that depends on a deprecated API. I'd be OK if we extracted this to a function that used the new behavior with PushKit if available and fell back to using deprecated API if not available.